hashexpr/src/atom.rs

```diff
@@ -50,11 +51,12 @@ impl Atom {
     }
   }
 
+  #[must_use]
   pub fn data_bytes(&self) -> Vec<u8> {
     match self {
       Self::Link(x) => x.as_bytes().to_vec(),
-      Self::Bits(x) => x.to_owned(),
-      Self::Text(x) => x.to_owned().into_bytes(),
+      Self::Bits(x) => x.clone(),
```

Comment:
`to_owned` in these places actually delegates to `clone`. For example ```rs
let s: &str = "hi world";
let string: String = s.to_owned();
```
This works and is valid. But if `s` is a `String`, then `to_owned` delegates to `clone` because its role is to create a new owned value.
`clone` is clearer than `to_owned` because it makes it clear we are actually creating a new value here and not making a new owned version of an already owned value.
https://rust-lang.github.io/rust-clippy/master/index.html#implicit_clone